elide-core/src/main/java/com/yahoo/elide/parsers/state/RelationshipTerminalState.java

```diff
@@ -82,9 +82,31 @@ public RelationshipTerminalState(PersistentResource record, String relationshipN
         return () -> Pair.of(HttpStatus.SC_OK, mapper.toJsonObject(doc));
     }
 
+    /**
+     * Handle PATCH requests for relationship updates.
+     * <p>
+     * If request doesn't modify relationship, return 204,
+     * otherwise, return either 200 or 204 based on user's configuration on Elide
+     *
+     * @param state the state containing request information
+     *
+     * @return the proper update status code
+     */
     @Override
     public Supplier<Pair<Integer, JsonNode>> handlePatch(StateContext state) {
-        return handleRequest(state, this::patch);
+        RequestScope requestScope = state.getRequestScope();
+        int updateStatusCode = requestScope.getUpdateStatusCode();
+
+        boolean relationshipChanged = patch(state.getJsonApiDocument().getData(), requestScope);
```

Comment:
I'm also not sure this is correct. We're not looking for *any* change we're looking for changes that differ from those sent by the client.